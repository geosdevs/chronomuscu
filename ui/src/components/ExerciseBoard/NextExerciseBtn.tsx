import { faForward } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

type NextExerciseBtnProps = {
  onNextExerciseClick: Function;
  setReadOnlyState: Function;
};

export default function NextExerciseBtn({
  onNextExerciseClick,
  setReadOnlyState,
}: NextExerciseBtnProps) {
  return (
    <a
      className="group ml-6 relative inline-block text-sm font-bold text-chinarose focus:outline-none focus:ring active:text-slate-500"
      href="/"
      onClick={(e) => {
        e.preventDefault();
        onNextExerciseClick();
        setReadOnlyState(true);
      }}
    >
      <span className="rounded-lg absolute inset-0 translate-x-0 translate-y-0 bg-chinarose transition-transform group-hover:translate-x-0.5 group-hover:translate-y-0.5"></span>

      <span className="rounded-lg relative block border border-chinarose hover:border-current bg-white px-2.5 py-1.5">
        {" "}
        Next <FontAwesomeIcon icon={faForward} />{" "}
      </span>
    </a>
  );
}
