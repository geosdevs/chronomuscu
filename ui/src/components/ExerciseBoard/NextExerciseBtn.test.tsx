import { ExerciseBoardData } from "../../app-types";
import renderer from "react-test-renderer";
import NextExerciseBtn from "./NextExerciseBtn";

describe("NextExerciseBtn", () => {
  it("should render", () => {
    const onNextExerciseClickMock = jest.fn();
    const component = renderer.create(
      <NextExerciseBtn onNextExerciseClick={onNextExerciseClickMock} />
    );
    const jsonTree = component.toJSON();

    expect(jsonTree).toMatchSnapshot();
    expect(onNextExerciseClickMock).not.toHaveBeenCalled();
  });
});
