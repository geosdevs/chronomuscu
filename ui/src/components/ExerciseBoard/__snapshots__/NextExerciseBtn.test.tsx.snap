// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextExerciseBtn should render 1`] = `
<a
  className="group ml-6 relative inline-block text-sm font-bold text-chinarose hover:text-ecru focus:outline-none active:text-slate-500 transition-colors duration-200"
  href="/"
  onClick={[Function]}
>
  <span
    className="rounded-md absolute inset-0 translate-x-0 translate-y-0 bg-ecru transition-transform group-hover:translate-x-0.5 group-hover:translate-y-0.5"
  />
  <span
    className="rounded-md relative block border border-chinarose hover:border-current bg-white px-2.5 py-1.5"
  >
     
    Next 
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-forward "
      data-icon="forward"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     
  </span>
</a>
`;
